### DISCUSSION TITLE
Installing on Mac

### DISCUSSION THREAD

**Discussion by LinKayser** [leap71/PicoGK#5] [General]
The OpenVDB library has a few dependencies that are likely not present on your Mac. I will see if I can move some of them to link statically, so we don't have such a number of libraries to install on the target systems. However, there is a relatively straightforward way to install all these dependencies.

Download the file Mac.sh from the PicoGKRuntime repo:

https://github.com/leap71/PicoGKRuntime/tree/main/Install_Dependencies

Or just use the attached ZIP.

[Mac.sh.zip](https://github.com/leap71/PicoGK/files/13165692/Mac.sh.zip)

Extract to a folder, and open Terminal. Navigate to the Folder and type **./Mac.sh**

This will install Homebrew (https://brew.sh/), which is used to install all the dependencies of openvdb. Depending on what you already have on your system, this may take a while, but afterwards you should be able to run PicoGK just fine.

Remember to copy picogk.1.0.dylib (https://github.com/leap71/PicoGK/blob/main/Runtime/picogk.1.0.dylib) to **/usr/local/lib**

If you don't have a "lib" in /usr/local, just create one (MacOS will ask you to verify that you know what you are doing).

We will create an installer in the future that will deal with all of that. In the meantime, it's the usual terminal hell that accompanies every open source project. It will only get easier from now on.

Happy Computational Engineering,
Lin

**Comment by philikai** [comment]
Trying the install on my M1 mac. 
From the other discussion on the [First Drop of PicoGK ](https://github.com/leap71/PicoGK/discussions/3) it seems like it will not run on M1 and M2 MacBooks. Is that correct? 

>> I followed the installation steps and copied the lib to the lib folder on mac. Visual Studio does not recognise the library. Didn't get the security warning either.

**Reply by LinKayser** [reply]
Hi Philipp, it works well on M1/M2 (that's what we use). Have you done the additional steps above (running Mac.sh) to install the dependencies?

Best,
Lin

**Reply by philikai** [reply]
Hi Lin,

fix: a restart is needed. 
Thx for the library! 

Looking forward to more things to come! 

Phil

**Reply by LinKayser** [reply]
Awesome. Great to hear. Things will only get easier from here.
Lin
